core/src/apps/management/recovery_device/__init__.py

```diff
@@ -43,27 +49,26 @@
             await show_pin_invalid(ctx)
             raise wire.PinInvalid("PIN invalid")
 
-    # set up pin if requested
-    if msg.pin_protection:
-        if msg.dry_run:
-            raise wire.ProcessError("Can't setup PIN during dry_run recovery.")
-        newpin = await request_pin_confirm(ctx, allow_cancel=False)
-        config.change_pin(pin_to_int(""), pin_to_int(newpin), None, None)
-
-    if msg.u2f_counter:
-        storage.device.set_u2f_counter(msg.u2f_counter)
-    storage.device.load_settings(
-        label=msg.label, use_passphrase=msg.passphrase_protection
-    )
+    if not msg.dry_run:
+        # set up pin if requested
+        if msg.pin_protection:
+            newpin = await request_pin_confirm(ctx, allow_cancel=False)
+            config.change_pin(pin_to_int(""), pin_to_int(newpin), None, None)
+
+        if msg.u2f_counter is not None:
+            storage.device.set_u2f_counter(msg.u2f_counter)
+        storage.device.load_settings(
+            label=msg.label, use_passphrase=msg.passphrase_protection
+        )
+
     storage.recovery.set_in_progress(True)
-    if msg.dry_run:
-        storage.recovery.set_dry_run(msg.dry_run)
+    storage.recovery.set_dry_run(bool(msg.dry_run))
```

Comment:
Note: This enables sending a RecoveryDevice message with dry_run = False to turn it off, which seems reasonable.

Reply:
indeed. but you'll also note that this is what the original code did before your mypy change.

Reply:
Yuy :man_facepalming: . Thanks for the fix then.